modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -4015,6 +4025,48 @@ public void testUpdateAsset() throws Exception {
 		Assert.assertEquals("john@liferay.com", assetEntry.getTitle());
 	}
 
+	@Test
+	public void testUpdateCommerceOrderSystemObjectWithJavaDelegateObjectValidationRule()
+		throws Exception {
+
+		try (Closeable closeable = _registerTestObjectValidationRuleEngine()) {
+			ExpandoTable expandoTable = ExpandoTestUtil.addTable(
+				PortalUtil.getClassNameId(CommerceOrder.class),
+				ExpandoTableConstants.DEFAULT_TABLE_NAME);
+
+			ExpandoTestUtil.addColumn(
+				expandoTable, "customFieldName", ExpandoColumnConstants.STRING);
+
+			Group group = GroupTestUtil.addGroup();
+
+			User user = UserTestUtil.getAdminUser(group.getCompanyId());
+
+			CommerceCurrency commerceCurrency =
+				CommerceCurrencyTestUtil.addCommerceCurrency(
+					group.getCompanyId());
+
+			CommerceChannel commerceChannel =
+				CommerceTestUtil.addCommerceChannel(
+					group.getGroupId(), commerceCurrency.getCode());
+
+			CommerceOrder commerceOrder = CommerceTestUtil.addB2CCommerceOrder(
+				user.getUserId(), commerceChannel.getGroupId(),
+				commerceCurrency);
+
+			ObjectDefinition objectDefinition = _fetchSystemObjectDefinition(
+				"CommerceOrder");
+
+			ObjectValidationRule objectValidationRule =
+				_addObjectValidationRule(
+					_javaDelegateEngineKey, objectDefinition, StringPool.BLANK);
+
+			_commerceOrderLocalService.updateCommerceOrder(commerceOrder);
```

Comment:
Should we use `CommerceOrderLocalServiceUtil.updateCommerceOrder(commerceOrder)` to follow the pattern used in the other test for the User system object?